Shape_detection/include/CGAL/Shape_detection/Region_growing/Polygon_mesh/Plane_face_region.h

```diff
@@ -0,0 +1,460 @@
+// Copyright (c) 2018 INRIA Sophia-Antipolis (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Florent Lafarge, Simon Giraudot, Thien Hoang, Dmitry Anisimov
+//
+
+#ifndef CGAL_SHAPE_DETECTION_REGION_GROWING_POLYGON_MESH_PLANE_FACE_REGION_H
+#define CGAL_SHAPE_DETECTION_REGION_GROWING_POLYGON_MESH_PLANE_FACE_REGION_H
+
+#include <CGAL/license/Shape_detection.h>
+
+// Internal includes.
+#include <CGAL/Shape_detection/Region_growing/internal/property_map.h>
+#include <CGAL/Shape_detection/Region_growing/internal/utils.h>
+#ifdef CGAL_SD_RG_USE_PMP
+#include <CGAL/Polygon_mesh_processing/compute_normal.h>
+#endif
+
+namespace CGAL {
+namespace Shape_detection {
+namespace Polygon_mesh {
+
+  /*!
+    \ingroup PkgShapeDetectionRGOnMesh
+
+    \brief Region type based on the quality of the least squares plane
+    fit applied to faces of a polygon mesh.
+
+    This class fits a plane, using \ref PkgPrincipalComponentAnalysisDRef "PCA",
+    to chunks of faces in a polygon mesh and controls the quality of this fit.
+    If all quality conditions are satisfied, the chunk is accepted as a valid region,
+    otherwise rejected.
+
```

Comment:
Doc needs to be updated. The region is based on the support plane of the seed face.